crates/sncast/src/starknet_commands/script/run.rs

```diff
@@ -272,136 +279,247 @@ impl<'a> ExtensionLogic for CastScriptExtension<'a> {
+    user_args: Vec<Vec<Arg>>,
+}
+
+impl SyscallPtrAccess for CastScriptRuntime<'_> {
+    fn get_mut_syscall_ptr(&mut self) -> &mut Relocatable {
+        self.starknet_runtime.get_mut_syscall_ptr()
+    }
+}
+
+impl ResourceTracker for CastScriptRuntime<'_> {
+    fn consumed(&self) -> bool {
+        self.starknet_runtime.consumed()
+    }
+
+    fn consume_step(&mut self) {
+        self.starknet_runtime.consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.starknet_runtime.get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.starknet_runtime.run_resources()
+    }
+}
+
+impl SignalPropagator for CastScriptRuntime<'_> {
+    fn propagate_system_call_signal(
+        &mut self,
+        selector: DeprecatedSyscallSelector,
+        vm: &mut VirtualMachine,
+    ) {
+        self.starknet_runtime
+            .propagate_system_call_signal(selector, vm);
+    }
+
+    fn propagate_cheatcode_signal(&mut self, selector: &str, inputs: &[Felt]) {
+        self.starknet_runtime
+            .propagate_cheatcode_signal(selector, inputs);
+    }
+}
+
+fn args_size(args: &[Arg]) -> usize {
+    args.iter().map(Arg::size).sum()
+}
+
+impl HintProcessorLogic for CastScriptRuntime<'_> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt>,
+    ) -> std::result::Result<(), HintError> {
+        let maybe_extended_hint = hint_data.downcast_ref::<Hint>();
+
+        match maybe_extended_hint {
+            Some(Hint::External(ExternalHint::AddRelocationRule { src, dst })) => {
+                vm.add_relocation_rule(
+                    extract_relocatable(vm, src)?,
+                    extract_relocatable(vm, dst)?,
+                )?;
+                Ok(())
+            }
+            Some(Hint::External(ExternalHint::WriteRunParam { index, dst })) => {
+                let index = get_val(vm, index)?.to_usize().expect("Got a bad index.");
+                let mut stack = vec![(cell_ref_to_relocatable(dst, vm), &self.user_args[index])];
+                while let Some((mut buffer, values)) = stack.pop() {
+                    for value in values {
+                        match value {
+                            Arg::Value(v) => {
+                                vm.insert_value(buffer, v)?;
+                                buffer += 1;
+                            }
+                            Arg::Array(arr) => {
+                                let arr_buffer = vm.add_memory_segment();
+                                stack.push((arr_buffer, arr));
+                                vm.insert_value(buffer, arr_buffer)?;
+                                buffer += 1;
+                                vm.insert_value(buffer, (arr_buffer + args_size(arr))?)?;
+                                buffer += 1;
+                            }
```

Comment:
Should it inserted twice?

Reply:
Seems so, see source https://github.com/starkware-libs/cairo/blob/ada0450439a36d756223fba88dfd1f266f428f0c/crates/cairo-lang-runner/src/casm_run/mod.rs#L1340

Reply:
That's fine then 👍